### PR TITLE
Flatten tensors bigger than 4D to 1D (binary only)

### PR DESCRIPTION
**Pull request by DenisVieriu97**

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 1]
```diff
@@ -59,7 +59,7 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
     output = output_.contiguous();
     needsCopyToOutput = true;
   // else, determine if this is an in-place operation on a view output
-  } else if (output_.is_view() && (self.is_alias_of(output_) || other.is_alias_of(output_))) {
+  } else if (output_.storage_offset() && (self.is_alias_of(output_) || other.is_alias_of(output_))) {
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/BinaryOps.mm:62]: Why this change is needed ?

**Review by kulinseth — APPROVED** [review]

File: aten/src/ATen/native/mps/operations/BinaryOps.mm (hunk 1 above)
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:62]: the change is not needed but the `is_view` is not reliable for check view ops

Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:62]: can you create this in a separate PR and add a test case for it ?

Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:62]: Yes, I'll separate it out

Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/BinaryOps.mm:62]: @DenisVieriu97 , I am merging this in as I need to update our master. Can you please create a Test case for the is_view commit.?
